src/pages/Quiz.jsx

```diff
@@ -0,0 +1,133 @@
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import ErrorMessage from "../components/ErrorMessage";
+import { useQuiz } from "../components/quiz-context";
+
+const Quiz = () => {
+  const { category } = useQuiz();
+  const navigate = useNavigate();
+
+  const [options, setOptions] = useState();
+  const [score, setScore] = useState(0);
+  const [selected, setSelected] = useState();
+  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
+  const [error, setError] = useState(false);
+  const [btnName, setBtnName] = useState("Next");
+  const displayQuestion = category.map((items) => items.questions);
+  const displayQuestion2 = displayQuestion[0];
+
+  useEffect(() => {
+    setOptions(
+      displayQuestion2 &&
+        handleShuffle([
+          displayQuestion2[currentQuestionIndex]?.correctAnswer,
+          ...displayQuestion2[currentQuestionIndex]?.wrongAnswer
+        ])
+    );
+  }, [displayQuestion2, currentQuestionIndex]);
+
+  const handleShuffle = (options) => {
+    return options.sort(() => Math.random() - 0.5);
+  };
+  const handleSelect = (opt) => {
+    if (
+      selected === opt &&
+      selected === displayQuestion2[currentQuestionIndex].correctAnswer
+    ) {
+      return "correct-answer";
+    } else if (
+      selected === opt &&
+      selected !== displayQuestion2[currentQuestionIndex].correctAnswer
+    ) {
+      return "wrong-answer";
+    } else if (opt === displayQuestion2[currentQuestionIndex].correctAnswer) {
+      return "correct-answer";
+    }
+  };
+
+  const handleCheck = (opt) => {
+    setSelected(opt);
+    if (opt === displayQuestion2[currentQuestionIndex].correctAnswer)
+      setScore(score + 10);
+    setError(false);
+  };
+
+  const handleNext = () => {
+    if (currentQuestionIndex === 3) {
+      setBtnName("Submit");
+    }
```

Comment:
Suggestion: I think state for button text is not needed,  you can do something like this 

`<button>{ currentQuestionIndex === 3 ? "Submit" : "Next" }</button>`